library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/DecorViewMapper.kt

```diff
@@ -27,27 +28,36 @@ internal class DecorViewMapper(
                 view
             )
         }
+        val decorClassName = view.javaClass.name
         // we will add the window wireframe here which comes in handy whenever we have to record a
         // pop - up. Usually the pop - up gets displayed in a smaller decorView in a upper Window.
         // the window behind is blurred so we need this extra Wireframe here to achieve that
-        // effect in the player.
-        val windowWireframeId = uniqueIdentifierGenerator.resolveChildUniqueIdentifier(
-            view,
-            WINDOW_KEY_NAME
-        )
-        if (windowWireframeId != null) {
-            val windowWireframe = MobileSegment.Wireframe.ShapeWireframe(
-                id = windowWireframeId,
-                x = 0,
-                y = 0,
-                width = systemInformation.screenBounds.width,
-                height = systemInformation.screenBounds.height,
-                shapeStyle = MobileSegment.ShapeStyle(
-                    backgroundColor = WINDOW_WIREFRAME_COLOR,
-                    opacity = WINDOW_WIREFRAME_OPACITY
+        // effect in the player. We will only do this for `DialogWindow@DecorView` and not for
+        // `PopUpWindow@PopUpDecorView` types in order to follow what the system does. We do not have
+        // a way to access that type because it is private in the Android system so we will try to assert
+        // the class name in this case. Quite a nasty workaround but given the limitations is the best
+        // we could find. We know that the system classes are not obfuscated by Proguard so in case
+        // this class name will be changed in the future we will see it in our replays and fix it.
+        if (!decorClassName.lowercase(Locale.ENGLISH)
```

Comment:
Not sure if we need lowercase though here, normally class names follow the convention.

```suggestion
        if (!decorClassName.lowercase(Locale.US)
```

Reply:
yeah...good point.